src/core/io/src/4C_io_input_spec_builders.hpp

```diff
@@ -1655,6 +1721,24 @@ Core::IO::InputSpec Core::IO::InputSpecBuilders::parameter(
       });
 }
 
+template <typename T>
+  requires(std::is_same_v<T, std::string>)
+Core::IO::InputSpec Core::IO::InputSpecBuilders::selection(
+    std::string name, Internal::BasedOn selector, Internal::FromChoices choices, GroupData data)
+{
+  std::vector<InputSpec> specs;
+  specs.reserve(choices.choices.size());
+  for (auto&& [choice_name, choice_spec] : choices.choices)
+  {
+    specs.emplace_back(all_of({
+        selection<std::string>(selector.name, {choice_name},
+            {.description = "Selects which other parameters are allowed in this group."}),
+        std::move(choice_spec),
+    }));
+  }
+  return group(name, {one_of(specs)}, data);
```

Comment:
@gilrrei For now, this implemented in terms of existing features. We can think about using a separate implementation internally to emit special info for this case. However, I would like to wait with that until we converged on the exact semantics of the new function.